routes/duties.js

```diff
@@ -0,0 +1,141 @@
+import {
+  createNewDuty, lookForAllDuties, lookForDutyById, deleteDutyById, updateDuty,
+} from '../database.js';
+
+const newDutySchema = {
+  schema: {
+    body: {
+      type: 'object',
+      additionalProperties: false,
+      required: ['name', 'location', 'time', 'constraints', 'soldiersRequired', 'value'],
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+
+    },
+  },
+
+};
+
+const getDutySchema = {
+  schema: {
+    query: {
+      type: 'object',
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+    },
+  },
+
+};
+
+const dutyEditSchema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    location: { type: 'string' },
+    time: {
+      type: 'object',
+      properties: {
+        start: { type: 'string', format: 'date' },
+        end: { type: 'string', format: 'date' },
+      },
+    },
+    constraints: { type: 'array' },
+    soldiersRequired: { type: 'number' },
+    value: { type: 'number' },
+  },
+
+};
+
+export default function dutyRouter(app, opts, done) {
+  app.post('/', newDutySchema, async (req, res) => {
+    const dutyInserted = req.body;
+    dutyInserted._id = Math.floor(Math.random() * 1000000);
+    await createNewDuty(dutyInserted);
+    res.status(201).send({ message: 'duty created' });
```

Comment:
how the client know what is the inserted id?

Reply:
fixed by returning it